### PR TITLE
Added Manhattan distance algorithm

### PR DESCRIPTION
**Pull request by carlos3dx**
### Describe your change:

Added an algorithm to calculate Manhattan distance,

Added the algorithm twice, one in a mor traditional way and other using a one-liner.

Fixes: #7776

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/manhattan_distance.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    TypeError: Expected a list of numbers as input, found int
+    >>> _validate_point("not_a_list")
+    Traceback (most recent call last):
+         ...
+    TypeError: Expected a list of numbers as input, found str
+    """
+    if point:
+        if isinstance(point, list):
+            for item in point:
+                if not (isinstance(item, int) or isinstance(item, float)):
```
Comment (cclauss) [inline maths/manhattan_distance.py:75]:
```suggestion
                if not isinstance(item, (int, float)):
```

Comment (carlos3dx) [reply maths/manhattan_distance.py]: Forgot that it can accept a tuple, fixed

File: maths/manhattan_distance.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    _validate_point(point_a)
+    _validate_point(point_b)
+    if len(point_a) != len(point_b):
+        raise ValueError("Both points must be in the same n-dimensional space")
+
+    total_distance = 0.0
+    for i in range(len(point_a)):
+        total_distance += abs(point_a[i] - point_b[i])
+
+    return total_distance
```
Comment (cclauss) [inline maths/manhattan_distance.py:50]:
```suggestion
    return sum(a - b for a, b in zip(point_a, point_b)) 
```

File: maths/manhattan_distance.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+
+def _validate_point(point: Any) -> None:
```
Comment (cclauss) [inline maths/manhattan_distance.py:53]: Why `Any`?!?  We want `list[int | float]`.

Comment (cclauss) [inline maths/manhattan_distance.py:53]:
```suggestion
def _validate_point(point: list[float]) -> None:
```

**Review by cclauss — APPROVED** [review]

File: maths/manhattan_distance.py (hunk 3 above)
Comment (carlos3dx) [reply maths/manhattan_distance.py]: Cause being a validation function I expect anything as input and inside I do the checks.

File: maths/manhattan_distance.py (hunk 2 above)
Comment (carlos3dx) [reply maths/manhattan_distance.py]: As I mentioned in the PR, I implemented twice, one with firs and one with a one liner that yo can see at the bottom.

File: maths/manhattan_distance.py (hunk 3 above)
Comment (cclauss) [reply maths/manhattan_distance.py]: That is not the way that type hints work.

**Pull request merged**